src/components/MediaViewer/MediaViewer.tsx

```diff
@@ -351,500 +366,615 @@ const MediaViewer = ({ resource }: { resource: CloudinaryResource }) => {
-          className="object-contain"
-          width={resource.width}
-          height={resource.height}
-          src={resource.public_id}
-          alt={`Image ${resource.public_id}`}
-          style={imgStyles}
-          version={version}
-          placeholderStyle="dark"
-          {...transformations}
-        />
+            {!isTrash && (
+              <li>
+                {process.env.NEXT_PUBLIC_PHOTOBOX_MODE !== "read-only" && (
+                  <Button
+                    variant="ghost"
+                    className="text-white"
+                    onClick={() => setDeletion({ state: "confirm" })}
+                  >
+                    <Trash2 className="h-6 w-6" />
+                    <span className="sr-only">Delete</span>
+                  </Button>
+                )}
+                {process.env.NEXT_PUBLIC_PHOTOBOX_MODE === "read-only" && (
+                  <TooltipProvider delayDuration={200}>
+                    <Tooltip>
+                      <TooltipTrigger
+                        className={`${buttonVariants({ variant: "ghost" })} text-zinc-500 hover:text-zinc-500 bg-transparent hover:bg-transparent`}
+                        aria-label="Deleting is disabled"
+                      >
+                        <Trash2 className="h-6 w-6" />
+                        <span className="sr-only">Delete</span>
+                      </TooltipTrigger>
+                      <TooltipContent>
+                        <p>Deleting is Disabled</p>
+                      </TooltipContent>
+                    </Tooltip>
+                  </TooltipProvider>
+                )}
+              </li>
+            )}
+            {isTrash && (
+              <li>
+                <Button
+                  variant="ghost"
+                  className="text-white"
+                  onClick={handleOnRestore}
+                >
+                  <History className="h-6 w-6" />
+                  <span className="sr-only">Restore</span>
+                </Button>
+              </li>
+            )}
+          </ul>
+        </Container>
+
+        {/** Asset viewer */}
+
+        <div className="relative flex justify-center items-center align-center w-full h-full">
+          <CldImage
+            key={`${JSON.stringify(transformations)}-${version}`}
+            className="object-contain"
+            width={resource.width}
+            height={resource.height}
+            src={resource.public_id}
+            alt={`Image ${resource.public_id}`}
+            style={imgStyles}
+            version={version}
+            placeholderStyle="dark"
+            {...transformations}
+          />
+        </div>
       </div>
-    </div>
-  )
+
+      {/** Confirmation dialog to display if there are unsaved edits */}
+
+      <Dialog open={showConfirmDialog}>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Save Changes?</DialogTitle>
```

Comment:
this actually confused me a bit. With the headline saying Save Changes, I automatically assume that Yes is going to save the changes for me, regardless of what the paragraph says, but it seems to be a bit conflicting

I would expect either:
- This says "Are you sure?" or something like that
- Yes would save first before going back, no would go back without saving

I think im okay with either

but fwiw, heres what Google Photos does

<img width="432" alt="image" src="https://github.com/cloudinary-community/photobox/assets/1045274/1cf50983-eb42-4c1e-91b1-5f4e78dc7c4d">

i think this also follows along with the UX of my other comment, where maybe it makes senes that if a panel is open, ESC closes the panel first, so to get back with a panel open, you would hit Esc twice

in that case, if the panel is open, and changes are made, the action would close the panel, but you're confirming whether or not you want to save the changes or not

perhaps we follow Google Photos?

wdyt?

Reply:
Sounds good! I'll make the change.

Reply:
Feedback addressed in 9e85431395bfd9897bf0b7fa40301f8785bf4f16